### ISSUE TITLE
allow label instead of `#`

### ISSUE THREAD

**Issue by cscheid** [quarto-dev/quarto-cli#7981]
This should work: (instead of `#lst-1`, etc)

````
```{.r label="lst-1" lst-cap="Here's a listing caption"}
plot(1:10)
```
````

Sometimes we type this by accident...

**Comment by cderv** [comment]
I think if we do allow that in this syntax, it would mean that we also need to supported elsewhere in `Div` `Spans` and any other place where Pandoc expect a `#` to be used for id. Or at least we open the door too. 

I understand though that we could explain this is scope to code cells only, because of the close similarity to computation cells. But `#` would still need to be learn by users to set ID on other elements. 

So I am not that convince that `#lst-1` is more complex than `label="lst-1"`. But I may be biased and relying on YAML autocompletion to much, that I rarely type the whole `label` part. 

As we don't have autocompletion on code cells, I usually go with the `#` road. We'll see if I am alone in that case 😅 😉

**Comment by cscheid** [comment]
> I think if we do allow that in this syntax, it would mean that we also need to supported elsewhere in Div Spans and any other place where Pandoc expect a # to be used for id. Or at least we open the door too.

Yes, totally. The easiest way to do it is to add a filter, very early on in our Lua processing, that converts all `label` keywords in divs and spans to identifiers.

> So I am not that convince that #lst-1 is more complex than label="lst-1". 

It's not more complex, I agree! But like I said in the issue, sometimes we type `label="lst-1"` by accident, and there's no error message or anything. It'd be nice if the syntax just worked.

**Comment by mcanouil** [comment]
Note that `id` works.

````qmd
---
title: "Untitled"
format: html
---

```{.r id="lst-1" lst-cap="Here's a listing caption"}
plot(1:10)
```
````

**Comment by cderv** [comment]
Interesting !! Nice finding ! 

This is Pandoc support parsing `id=` as `#` which makes sense when you think about it
````
> pandoc -t native 
```{.r id="lst-1" lst-cap="Here's a listing caption"}
plot(1:10)
```
^Z
[ CodeBlock
    ( "lst-1"
    , [ "r" ]
    , [ ( "lst-cap" , "Here's a listing caption" ) ]
    )
    "plot(1:10)"
]
````

**Comment by mcanouil** [comment]
I thought you knew^^
Any valid HTML attributes are taken as-is, thus `id` working.

**Comment by cderv** [comment]
> Any valid HTML attributes are taken as-is, thus id working.

If you are referring to setting attributes on div or else, and them being passed as-is to HTML, I don't think it is exactly the same here. 

All the attributes are parsed in Pandoc AST with the `key=value`, without transformation:
````
❯ pandoc --to native
```{.r dummy=something}
content
```
^Z
[ CodeBlock
    ( "" , [ "r" ] , [ ( "dummy" , "something" ) ] ) "content"
]
````

So for `id=lst-1`, I would have expected 
````
[ CodeBlock
    ( "" , [ "r" ] , [ ( "id" , "lst-1" ) ] ) "content"
]
````

But as `id` is a special name for Pandoc, they do consider `id=value` the same as using the usual syntax `#`

It is in fact exactly the same for class
````
❯ pandoc --to native
```{class=r}
1 + 1
```
^Z
[ CodeBlock ( "" , [ "r" ] , [] ) "1 + 1" ]


https://github.com/jgm/pandoc/blob/48f881d74620beddaa6f4c96c2239d179089b059/src/Text/Pandoc/Readers/Markdown.hs#L655-L668
